cmd/bcpd/app/app.go

```diff
@@ -48,6 +48,7 @@ func Chain(authFn x.Authenticator) app.Decorators {
 		utils.NewSavepoint().OnCheck(),
 		sigs.NewDecorator(),
 		multisig.NewDecorator(authFn),
+		// note: this also embeds utils.NewSavepoint().OnDeliver()
```

Comment:
I think this comment would be better as 

> cash.NewDynamicFeeDecorator embeds utils.NewSavepoint().OnDeliver() and therefore [this is missing].

- no need for "note:" as this is comment. No need to state the obvious,
- it is better to avoid context dependant words in short messages. Instead of "this" I think it is better to write the full name of what you have in mind. This reduce the chance of a mistake.
- a more board explanation is missing, hence my `[this is missing]`. Right now, I do not understand why this information is important :confused: 

I know this is a very picky comment. I think it is important to maintain comments that explain the code. It is hard task and if not paying enough care the result might be a comment that introduce more confusion than it helps.

Reply:
I'm not actually sure any of this makes any difference, in fact my concern is that I'd like to avoid these notes altogether and use `utils.NewSavepoint().OnDeliver()` instead of it being embedded in the DynamicFeeDecorator. 

I'm going to wait for Ethan's feedback on this one to see if we can resolve it somehow else.

Reply:
And yes, this is needed because some of the extensions depend on the fact that there's a checkpoint somewhere upstream. An example of such extension is `aswap`. And for any user that would want to assemble their own app/extension it would be very important to know whether this is present or not.

Reply:
I agree with Piotr with the formation of the sentence;

> cash.NewDynamicFeeDecorator embeds utils.NewSavepoint().OnDeliver()

I think we need to explain why we use Checkpoints in another document, not here, but this explains why the decorator is not in the stack